spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -173,6 +179,30 @@ sealed class SpdxExpression {
         return this
     }
 
+    /**
+     * Apply [licenseChoices] in the given order to [this][SpdxExpression].
+     */
+    fun applyChoices(licenseChoices: List<LicenseChoice>): SpdxExpression {
+        if (validChoices().size == 1) return this
+
+        var currentExpression = this
+
+        licenseChoices.forEach {
+            currentExpression = if (it.given == null) {
+                // If choice is not applicable, an InvalidLicenseChoiceException is thrown.
```

Comment:
What's the reason for making this function throw an exception? 
Have you considered just 'not applying' choices which are not applicable?
(As license may change, license choices may become invalid.The effect of not applying a choice will be propagated by the final applicable license expression which will then contain an OR. If it doesn't contain an OR, there also shouldn't be a reason why an exception is needed. What do you think?

Reply:
Refactored the code, now it's no longer throwing an exception, but rather checking in `LicenseChoice`s init function if the user misconfigured the license choice.

Reply:
Have you pushed that new init function already? (I can't find it)

Reply:
It is pushed now